### PR TITLE
Meta SNIP-25: security council

### PR DESCRIPTION
**Pull request by Eikix**
Drafting the meta SNIP for Security Council as part of the [Fête du SNIP](https://lu.ma/LeFeteduSNIP?locale=fr&tk=vmF03x)

### CODE REVIEW

**Comment by lucadonnoh** [comment]
For reference, this is the current upgradability structure:

image

Starknet would need to give up the permissions for all Proxy governors, Implementation governors, Verifier governors and bridge owners. Also, as per the [Risk rosette framework](https://gov.l2beat.com/t/the-risk-rosette-framework/292) and the Stages framework, the Exit window and Stage designation will stay at zero until forced transactions are implemented.

File: SNIPS/snip-75.md
Comment (Eikix) [reply SNIPS/snip-75.md:56]: I meant:
If the rollup contract is controlled by SW multisig but has timelock
And SC multisig is controlled by its members and has special right on the rollup contract

Comment (lucadonnoh) [reply SNIPS/snip-75.md:56]: yes. but the members shouldn't be reelected by the foundation, so last line should be changed

Comment (Eikix) [reply SNIPS/snip-75.md:56]: > yes. but the members shouldn't be reelected by the foundation, so last line should be changed

how do they get reelected?

Comment (lucadonnoh) [reply SNIPS/snip-75.md:56]: as said in the first comment, either via an onchain election mechanism like Arbitrum or the SC updates itself

Comment (Eikix) [reply SNIPS/snip-75.md]: Ok changing

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:4]: Need to add one last author: luiz-lvj <lvj253@gmail.com>
Waiting to get his full name

**Comment by Eikix** [comment]
Could we get Starkware's opinion and more clarity on the roadmap, @iliav-starkware? (Can't ping Boaz, though I know he owns this track)

**Comment by leo-starkware** [comment]
Hey @Eikix, could you add sections "Rationale" and "Security considerations" to the SNIP, in order to keep a uniform syntax among SNIPs? Let's merge afterwards (after renumbering the SNIP)

**Comment by Eikix** [comment]
> Hey @Eikix, could you add sections "Rationale" and "Security considerations" to the SNIP, in order to keep a uniform syntax among SNIPs? Let's merge afterwards (after renumbering the SNIP)

before merging, could we get Boaz's input or whoever is in charge of the security council is Starknet?
we could get the views of core devs of Starknet, since Security Council triggers downstream big changes to the L1 contracts of Starknet

**Comment by Eikix** [comment]
> There hasn't been any activity on this pull request recently, and in order to prioritize active work, it has been marked as stale. This PR will be closed and locked in 7 days if no further activity occurs. Thank you for your contributions!

Deactivating Stale Label.

I've heard Starkware (SW) and Starknet Foundation (SNF) have iterated and found an alternative design for the Security Council.
Can someone from SW or SNF modify this Pull Request and directly add their design here?

We'll then be able to merge the SNIP and move forward

**Comment by leo-starkware** [comment]
@BoazStark could you address @Eikix 's comment above? I'm not aware of the developments

**Comment by nconsigny** [comment]
> I've heard Starkware (SW) and Starknet Foundation (SNF) have iterated and found an alternative design for the Security Council. Can someone from SW or SNF modify this Pull Request and directly add their design here?

I've helped the Starknet Foundation with this one and I merged the design with the existing proposal, let me know what you think @Eikix

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:4]: main author should be whoever designed the document from SW / SNF since I was not involved

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:101]: Where is this amount decided? Based on what? Is it public?

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:145]: removed by whom? how?

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:176]: revoking allowance of what contract

**Review by Eikix — COMMENTED** [review]
There is no mention of an actor being able to pause Starknet. Is this on purpose?

File: SNIPS/snip-75.md
Comment (nconsigny) [reply SNIPS/snip-75.md:101]: An amount was decided it should be public indeed I just didn't feel like it belonged to the PR 😅

File: SNIPS/snip-75.md
Comment (nconsigny) [reply SNIPS/snip-75.md:4]: I merged both document trying to keep your doc as a basis

**Comment by nconsigny** [comment]
@Eikix updated with the poause flow and a few modifications, wdyt ?

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:4]: you can set Henri and Boaz as first writers, wdyt

File: SNIPS/snip-75.md
Comment (Eikix) [inline SNIPS/snip-75.md:77]: Does the pause inherently have an expiration?
Like pause until unpause or at most for 1 day

**Comment by Eikix** [comment]
Overall good! what are the next steps wrt SNIP?

File: SNIPS/snip-75.md
Comment (l-henri) [reply SNIPS/snip-75.md:75]: There is no automatic unpause in the current design

File: SNIPS/snip-75.md
Comment (l-henri) [reply SNIPS/snip-75.md:199]: The security agent is the entity that can pause the affected contract. 
See the pausing flow https://github.com/Eikix/SNIPs/blob/security-council/SNIPS/snip-75.md#pausing-flow

**Comment by Eikix** [comment]
changed number to 25 as mentioned by @leo-starkware

File: SNIPS/snip-25.md
Comment (leo-starkware) [inline SNIPS/snip-25.md:12]: surround

File: SNIPS/snip-25.md
Comment (leo-starkware) [inline SNIPS/snip-25.md:16]: The whole line 14 appears in section 'Motivation'. Could you delete it from the abstract?

**Review by leo-starkware — CHANGES_REQUESTED** [review]

File: SNIPS/snip-25.md
Comment (l-henri) [reply SNIPS/snip-25.md]: Done!

File: SNIPS/snip-25.md
Comment (l-henri) [reply SNIPS/snip-25.md:14]: Done!

**Review by l-henri — APPROVED** [review]
I think it's good to go!

**Pull request merged**
